src/TextArea.tsx

```diff
@@ -0,0 +1,164 @@
+import styled, {css} from 'styled-components'
+import React, {TextareaHTMLAttributes, ReactElement} from 'react'
+import {FormValidationStatus} from './utils/types/FormValidationStatus'
+import sx, {SxProp} from './sx'
+import {get} from './constants'
+
+export type TextAreaProps = {
+  /**
+   * Apply inactive visual appearance to the TextArea
+   */
+  disabled?: boolean
+  /**
+   * Indicates whether the TextArea must be checked
+   */
+  required?: boolean
+  /**
+   * Indicates whether the TextArea validation state
+   */
+  validationStatus?: FormValidationStatus
+  /**
+   * Block
+   */
+  block?: boolean
+  /**
+   * Allows resizing of the textarea
+   */
+  resize?: 'none' | 'both' | 'horizontal' | 'vertical'
+  /**
+   * Applies a custom width to the textarea
+   */
+  width?: number
```

Comment:
We don't have a `width` prop for with Select or TextInput. Why would we have it for TextArea?